pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -353,21 +353,24 @@ func (c *CloudPrivateIPConfigController) SyncHandler(key string) error {
 func (c *CloudPrivateIPConfigController) updateCloudPrivateIPConfigStatus(cloudPrivateIPConfig *cloudnetworkv1.CloudPrivateIPConfig, status *cloudnetworkv1.CloudPrivateIPConfigStatus) (*cloudnetworkv1.CloudPrivateIPConfig, error) {
 	updatedCloudPrivateIPConfig := &cloudnetworkv1.CloudPrivateIPConfig{}
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		ctx, cancel := context.WithTimeout(c.ctx, controller.ClientTimeout)
-		defer cancel()
-
 		// Get the latest item from the store before updating. We risk updating
 		// and overriding any client changes if we don't get the latest state of
 		// the object and update the status on that item. Ex: if we are in the
 		// middle of updating the status on item v1, after an update to the
 		// object v2 has been performed we'll override the changes made to v2
 		// back to v1 when updating below (because cloudPrivateIPConfig above
 		// will have the old state before any potential client changes).
-		cloudPrivateIPConfigLatest, err := c.cloudNetworkClient.CloudV1().CloudPrivateIPConfigs().Get(ctx, cloudPrivateIPConfig.Name, metav1.GetOptions{})
+		cloudPrivateIPConfigLatest, err := c.getCloudPrivateIPConfig(cloudPrivateIPConfig.Name)
```

Comment:
This doesn't seem right to me; you should pass this function's context down to `getCloudPrivateIPConfig`

Reply:
Why is that needed? `getCloudPrivateIPConfig` uses the global timeout defined for all kube client calls and can initiate it's own context based on that.